include/mbedtls/check_config.h

```diff
@@ -333,13 +333,17 @@
 #endif
 #undef MBEDTLS_HAS_MEMSAN
 
-#if defined(MBEDTLS_CCM_C) && (                                        \
-    !defined(MBEDTLS_AES_C) && !defined(MBEDTLS_CAMELLIA_C) && !defined(MBEDTLS_ARIA_C) )
+#if defined(MBEDTLS_CCM_C) && \
+    !(defined(MBEDTLS_BLOCK_CIPHER_CAN_AES) || defined(MBEDTLS_AES_C) || \
+    defined(MBEDTLS_BLOCK_CIPHER_CAN_ARIA) || defined(MBEDTLS_ARIA_C) || \
+    defined(MBEDTLS_BLOCK_CIPHER_CAN_CAMELLIA) || defined(MBEDTLS_CAMELLIA_C))
```

Comment:
Note: this could be make slightly shorter by using `MBEDTLS_CCM_GCM_CAN_CAMELLIA` & Co.